bpf/lib/drop.h

```diff
@@ -105,12 +105,8 @@ _send_drop_notify(__u8 file, __u16 line, struct __ctx_buff *ctx,
 	    !__builtin_constant_p(line) || line > 0xffff)
 		__throw_build_bug();
 
-	/* These fields should be constants, and non-zero 'dst_id' is only to be
-	 * used for ingress.
-	 */
-	if (!__builtin_constant_p(dst_id) ||
-	    (dst_id != 0 &&
-	     (!__builtin_constant_p(direction) || direction != METRIC_INGRESS)))
+	/* Non-zero 'dst_id' is only to be used for ingress. */
+	if (dst_id != 0 && (!__builtin_constant_p(direction) || direction != METRIC_INGRESS))
 		__throw_build_bug();
```

Comment:
I wonder if a build bug can still be thrown if the evaluation can not be done at build time, and conversely how the build bug will behave at runtime, if executed?

Given that the constraint is not violated then this is likely not going to cause problems, but it would be interesting to know why the dst_id is no longer a build time constant?

Maybe we could do a clang version check and only get rid of the contant check for the affected clang version for now?

Reply:
> Maybe we could do a clang version check and only get rid of the contant check for the affected clang version for now?

Good point :100: 

Once I have some free time, I might just dig around to check why it's failing for clang 14 and above.